src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java

```diff
@@ -145,33 +140,38 @@ private RSAPrivateKey readPrivateKey(String file) throws IOException {
         }
     }
 
-    @Override
-    protected void configure(final HttpSecurity http) throws Exception {
+    /**
+     * Since this configuration is annotated with {@link Order} and {@link SecurityConfiguration}
+     * is not, this configuration is evaluated first when the SAML2 Profile is active.
+     *
+     * @param http The Spring http security configurer.
+     * @return The configured http security filter chain.
+     * @throws Exception Thrown in case Spring detects an issue with the security configuration.
+     */
+    @Bean
+    @Order(1)
+    protected SecurityFilterChain saml2FilterChain(final HttpSecurity http) throws Exception {
         // @formatter:off
         http
-            .requestMatchers()
-                // This filter chain is only applied if the URL matches
-                // Else the request is filtered by {@link SecurityConfiguration}.
-                .antMatchers("/api/public/saml2")
-                .antMatchers("/saml2/**")
-                .antMatchers("/login/**")
-            .and()
-            .csrf()
-                // Needed for SAML to work properly
-                .disable()
-            .authorizeRequests()
+            // This filter chain is only applied if the URL matches
+            // Else the request is filtered by {@link SecurityConfiguration}.
+            .securityMatcher("/api/saml2", "/saml2/**", "/login/**")
```

Comment:
why is "/api/public/saml2" not needed anymore?

Reply:
Very good catch. That's a faulty merge. Gonna revert